src/main/java/org/openstreetmap/atlas/checks/distributed/IntegrityChecksCommandArguments.java

```diff
@@ -0,0 +1,189 @@
+package org.openstreetmap.atlas.checks.distributed;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.openstreetmap.atlas.checks.atlas.CountrySpecificAtlasFilePathFilter;
+import org.openstreetmap.atlas.checks.base.Check;
+import org.openstreetmap.atlas.checks.constants.CommonConstants;
+import org.openstreetmap.atlas.checks.maproulette.MapRouletteConfiguration;
+import org.openstreetmap.atlas.generator.tools.filesystem.FileSystemHelper;
+import org.openstreetmap.atlas.generator.tools.spark.SparkJob;
+import org.openstreetmap.atlas.geography.Rectangle;
+import org.openstreetmap.atlas.geography.atlas.Atlas;
+import org.openstreetmap.atlas.geography.atlas.items.AtlasEntity;
+import org.openstreetmap.atlas.geography.atlas.items.AtlasObject;
+import org.openstreetmap.atlas.geography.sharding.Shard;
+import org.openstreetmap.atlas.geography.sharding.SlippyTile;
+import org.openstreetmap.atlas.utilities.collections.Iterables;
+import org.openstreetmap.atlas.utilities.collections.MultiIterable;
+import org.openstreetmap.atlas.utilities.collections.StringList;
+import org.openstreetmap.atlas.utilities.configuration.Configuration;
+import org.openstreetmap.atlas.utilities.conversion.StringConverter;
+import org.openstreetmap.atlas.utilities.maps.MultiMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Handles arguments and base functionality for integrity check sparkjobs generating commands
+ *
+ * @author jklamer
+ */
+public abstract class IntegrityChecksCommandArguments extends SparkJob
+{
+    /**
+     * @author brian_l_davis
+     */
+    protected enum OutputFormats
+    {
+        FLAGS,
+        GEOJSON,
+        METRICS,
+        TIPPECANOE
+    }
+
+    @Deprecated
+    protected static final Switch<String> ATLAS_FOLDER = new Switch<>("inputFolder",
```

Comment:
Why is this deprecated?

Reply:
That is a carryover from @jklamer's original PR. My assumption, and why I left it, is that it would be better to use the 'input' switch from the SparkJob base class.